grafana-plugin/pkg/plugin/install.go

```diff
@@ -104,33 +83,66 @@ func (a *App) handleInstall(w http.ResponseWriter, req *http.Request) {
 			}
 		}
 
-		w.Header().Add("Content-Type", "application/json")
-		if err := json.NewEncoder(w).Encode(installError); err != nil {
-			http.Error(w, "Failed to encode response", http.StatusInternalServerError)
-			return
-		}
-		w.WriteHeader(http.StatusBadRequest)
+		return nil, http.StatusBadRequest, &installError, nil
 	} else {
 		provisionBody, err := io.ReadAll(res.Body)
 		if err != nil {
 			log.DefaultLogger.Error("Error reading response body", "error", err)
-			return
+			return nil, http.StatusInternalServerError, nil, err
 		}
 
-		var provisioningData OnCallProvisioningJSONData
 		err = json.Unmarshal(provisionBody, &provisioningData)
 		if err != nil {
 			log.DefaultLogger.Error("Error unmarshalling OnCallProvisioningJSONData", "error", err)
-			return
+			return nil, http.StatusInternalServerError, nil, err
 		}
 
-		onCallPluginSettings.OnCallToken = provisioningData.OnCallToken
-		err = a.SaveOnCallSettings(onCallPluginSettings)
-		if err != nil {
-			log.DefaultLogger.Error("Error saving settings", "error", err)
-			return
+		if persistPluginSettingsToGrafana {
+			pluginSettings.OnCallToken = provisioningData.OnCallToken
+
+			if err = a.SaveOnCallSettings(pluginSettings); err != nil {
+				log.DefaultLogger.Error("Error saving settings", "error", err)
+				return nil, http.StatusInternalServerError, nil, err
+			}
```

Comment:
similar comment to [this](https://github.com/grafana/oncall/pull/4987/files#r1744601876)